### PR TITLE
Fixes `httpcore` import in `httpx_ws/_api.py`

### LINKED ISSUE #35: `ModuleNotFoundError: No module named 'httpcore.backends'`
## Describe the bug

Trying to run a different project [jpterm](https://github.com/davidbrochart/jpterm) and encountered the following traceback:

```bash
Traceback (most recent call last):
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/bin/jpterm", line 5, in <module>
    from jpterm.cli import main
  File "/Users/samforeman/projects/jpterm/jpterm/cli.py", line 3, in <module>
    from txl.app import disabled
  File "/Users/samforeman/projects/jpterm/txl/txl/app.py", line 6, in <module>
    components = {
  File "/Users/samforeman/projects/jpterm/txl/txl/app.py", line 7, in <dictcomp>
    ep.name: ep.load()
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2517, in load
    return self.resolve()
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2523, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/lib/python3.10/site-packages/txl_remote_contents/components.py", line 9, in <module>
    from httpx_ws import aconnect_ws
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/lib/python3.10/site-packages/httpx_ws/__init__.py", line 3, in <module>
    from httpx_ws._api import (
  File "/Users/samforeman/projects/saforem2/l2hmc-qcd/venvs/py310/lib/python3.10/site-packages/httpx_ws/_api.py", line 14, in <module>
    from httpcore.backends.base import AsyncNetworkStream, NetworkStream
ModuleNotFoundError: No module named 'httpcore.backends'
```

I've identified the problem to be coming from this line in your [`httpx_ws/_api.py:L14`](./httpx_ws/_api.py:L14) file

https://github.com/frankie567/httpx-ws/blob/f0e4d4b124d3ca7b0a85178f8294c22542841ce5/httpx_ws/_api.py#L14

Replacing 

```python
from httpcore.backends.base import AsyncNetworkStream, NetworkStream
```

with 

```python
from httpcore._backends.base import AsyncNetworkStream, NetworkStream
```

resolved the issue.

## Configuration
 - Python version:
 - httpx-ws version:

### PR DESCRIPTION
**Pull request by saforem2**
Fixes #35

### CODE REVIEW

**Comment by frankie567** [comment]
Thank you @saforem2 for the fix!

So, it seems that we also need to do that here:

https://github.com/frankie567/httpx-ws/blob/f0e4d4b124d3ca7b0a85178f8294c22542841ce5/httpx_ws/transport.py#L8

I would also like to bump the dependency in `pyproject.toml`: `httpcore>=0.17.3,<0.18` So we're sure everyone is on the same level + avoid future breaking changes.

**Comment by saforem2** [comment]
should be good to go 🚀

File: httpx_ws/transport.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import anyio
 import wsproto
-from httpcore.backends.base import AsyncNetworkStream
+from httpcore import AsyncNetworkStream, NetworkStream
```
[the remark this answers is not in the document]
Comment (saforem2) [reply httpx_ws/transport.py]: lol all good, should be fixed now 👍

**Comment by saforem2** [comment]
ope, missed the import in [`tests/test_api.py`](https://github.com/frankie567/httpx-ws/blob/main/tests/test_api.py)

[83bfdb3](https://github.com/frankie567/httpx-ws/pull/36/commits/83bfdb3541ff7c4f9b4920013e56f664036783d6) should fix and resolve failing tests

**Comment by frankie567** [comment]
Lol, this one was trickier than expected! Good to go, now. Thank you for your efforts 🙏 

@all-contributors add @saforem2 for bug

**Pull request merged**